### PR TITLE
remove SwiftSyntax support for `_forget`, the old spelling of `discard`

### PR DESCRIPTION
**Pull request by kavon**
resolves rdar://112549258

paired with https://github.com/apple/swift/pull/67902

### CODE REVIEW

**Comment by kavon** [comment]
@swift-ci please test

File: CodeGeneration/Sources/generate-swiftsyntax/LayoutNode+Extensions.swift [hunk 1]
```diff
@@ -84,13 +84,13 @@ extension LayoutNode {
...
-        - trailingTrivia: Trivia to be appended to the trailing trivia of the node’s last token. \
-      If the node is empty, there is no token to attach the trivia to and the parameter is ignored.
-      """.removingEmptyLines
+    - Parameters:
+      - leadingTrivia: Trivia to be prepended to the leading trivia of the node’s first token. \
+    If the node is empty, there is no token to attach the trivia to and the parameter is ignored.
+    \(children.compactMap(generateParamDocComment).joined(separator: "\n"))
+      - trailingTrivia: Trivia to be appended to the trailing trivia of the node’s last token. \
+    If the node is empty, there is no token to attach the trivia to and the parameter is ignored.
+    """.removingEmptyLines
```
Comment (ahoppen) [inline CodeGeneration/Sources/generate-swiftsyntax/LayoutNode+Extensions.swift:93]: I suspect this is because you have a `swift-format` installed on your system that wasn’t built from `main`. I’m working on fixing that in principle in https://github.com/apple/swift-syntax/pull/2022 but for now can you just revert the changes to these lines?

**Review by ahoppen — APPROVED** [review]
Thanks

File: CodeGeneration/Sources/generate-swiftsyntax/LayoutNode+Extensions.swift (hunk 1 above)
Comment (kavon) [reply CodeGeneration/Sources/generate-swiftsyntax/LayoutNode+Extensions.swift:93]: oh. yeah sure thanks!

**Comment by kavon** [comment]
@swift-ci please test
